src/main/java/io/vertx/ext/camel/impl/CamelToVertxProcessor.java

```diff
@@ -54,30 +54,47 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
     Message in = exchange.getIn();
 
     Object body = CamelHelper.convert(inbound, in);
+
     DeliveryOptions delivery = CamelHelper.getDeliveryOptions(in, inbound.isHeadersCopy());
 
     try {
       if (inbound.isPublish()) {
-        vertx.eventBus().publish(inbound.getAddress(), body.toString(), delivery);
+        try {
+          vertx.eventBus().publish(inbound.getAddress(), body, delivery);
+        } catch (Exception e) {
+          // Mainly there to catch the missing codec exception
```

Comment:
If the missing codec exception is thrown by vertx before publishing the message then you should not catch it here, but let it catch at the end of all this block of code, so you return true from the method. As there is no async callback event from vert.x later that would trigger the continued routing on Camel.